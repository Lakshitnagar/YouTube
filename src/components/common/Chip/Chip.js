import React from 'react';
import PropTypes from "prop-types";
import './Chip.scss';

function Chip(props) {
    return (
        <div className="chip" onClick={()=>props.onClick(props.text)}>
            {props.text}
        </div>
    );
}

Chip.propTypes = {
    text: PropTypes.string,
    onClick: PropTypes.func
};

export default Chip;
