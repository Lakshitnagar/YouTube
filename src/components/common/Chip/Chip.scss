.chip{
  border: solid 1px rgba(0,0,0,0.1);
  height: 32px;
  line-height: 32px;
  padding: 0 12px;
  color: #030303;
  background-color: rgba(0,0,0,0.05);
  border-radius: 16px;
  cursor: pointer;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 200px;

  &:hover{
    background-color: rgba(0,0,0,0.1);
  }
}
