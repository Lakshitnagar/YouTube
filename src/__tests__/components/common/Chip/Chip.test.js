import React from 'react';
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import Chip from "../../../../components/common/Chip/Chip";

describe('Chip', function () {
    it('should render and match snapshot', function () {
        const component = <Chip/>;

        const wrapper = shallow(component);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should show text passed to it inside chip', function () {
        const component = <Chip text={'chip text'}/>;
        const wrapper = shallow(component);

        expect(wrapper.find('.chip').text()).toEqual('chip text');
    });

    it('should call onClick on clicking the chip with text that passed to it', function () {
        const mockOnChipClick = jest.fn();
        const component = <Chip text={'chip text'} onClick={mockOnChipClick}/>;
        const wrapper = shallow(component);

        wrapper.find('.chip').simulate('click');
        expect(mockOnChipClick).toHaveBeenCalledWith('chip text');
    });
});
