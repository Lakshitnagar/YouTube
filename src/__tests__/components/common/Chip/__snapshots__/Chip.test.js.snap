// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render and match snapshot 1`] = `
<div
  className="chip"
  onClick={[Function]}
/>
`;
